kernel: bump 5.10 to 5.10.70

rsalvaterra opened the pull request:
Deleted (upstreamed):
bcm27xx/patches-5.10/950-0312-vc4_hdmi-Set-HD_CTL_WHOLSMP-and-HD_CTL_CHALIGN_SET.patch
bcm53xx/patches-5.10/181-Revert-USB-bcma-Add-a-check-for-devm_gpiod_get.patch
generic/backport-5.10/610-v5.13-02-netfilter-Fix-fall-through-warnings-for-Clang.patch
generic/backport-5.10/792-v5.15-0001-net-dsa-b53-Fix-calculating-number-of-switch-ports.patch
generic/backport-5.10/792-v5.15-0002-net-dsa-b53-Set-correct-number-of-ports-in-the-DSA-s.patch
generic/backport-5.10/792-v5.15-0003-net-dsa-b53-Fix-IMP-port-setup-on-BCM5301x.patch
generic/backport-5.10/840-0001-PCI-of-Don-t-fail-devm_pci_alloc_host_bridge-on-miss.patch
generic/backport-5.10/840-0002-PCI-iproc-Fix-BCMA-probe-resource-handling.patch
generic/pending-5.10/498-mtd-mtdconcat-select-readwrite-function.patch

Manually refreshed patches:
generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch 
bcm27xx/patches-5.10/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch
bcm27xx/patches-5.10/950-0087-firmware-raspberrypi-Add-backward-compatible-get_thr.patch

Run-tested:
ath79/generic (TL-WDR3600) - .69
mvebu/cortexa9 (Turris Omnia) - .69
ramips/mt7621 (Redmi AC2100) - .70

Signed-off-by: Rui Salvaterra \<rsalvaterra@gmail.com\>

rsalvaterra committed:
kernel: bump 5.10 to 5.10.65

Manually refreshed:
generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch
bcm27xx/patches-5.10/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch
bcm27xx/patches-5.10/950-0087-firmware-raspberrypi-Add-backward-compatible-get_thr.patch

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

graysky2 commented:
You might want to replace `platform` with the actual value.  For example, `bcm27xx/patches-5.10/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch`

graysky2 commented:
Build system: x86_64
Build-tested: bcm2711/RPi4B, ipq806x/R7800
Run-tested: bcm2711/RPi4B, ipq806x/R7800

nicklowe commented:
I suggest that this is dropped and revisited with 5.10.66 due to the regressions in 5.10.65

rsalvaterra committed:
kernel: bump 5.10 to 5.10.66

No patches refreshed.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra commented:
> I suggest that this is dropped and revisited with 5.10.66 due to the regressions in 5.10.65

Check again… ;)

graysky2 commented:
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, ipq806x/R7800
Run-tested: bcm2711/RPi4B, ipq806x/R7800
```

nicklowe commented:
> > I suggest that this is dropped and revisited with 5.10.66 due to the regressions in 5.10.65
> 
> 
> 
> Check again… ;)

Looks good to me! :)

rockdrilla commented:
Gonna refresh again - 5.10.67 is out :sweat_smile:

graysky2 commented:
Just remove upstreamed patch (`bcm27xx/patches-5.10/950-0312-vc4_hdmi-Set-HD_CTL_WHOLSMP-and-HD_CTL_CHALIGN_SET.patch`) and autorefresh does the rest.

Ref: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.67&id=b32d3ded9dc2987edbc3427e66238bfdfcecf0ad

rsalvaterra commented:
> Gonna refresh again - 5.10.67 is out 😅

Indeed. Working on it… :)

rsalvaterra committed:
kernel: bump 5.10 to 5.10.67

Deleted (upstreamed):
bcm27xx/patches-5.10/950-0312-vc4_hdmi-Set-HD_CTL_WHOLSMP-and-HD_CTL_CHALIGN_SET.patch

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

graysky2 commented:
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, ipq806x/R7800
Run-tested: bcm2711/RPi4B, ipq806x/R7800
```

lister-wrt commented:
Thanks. No issues on x86_64 (apu4). Uptime ~16 hours

rsalvaterra commented:
Oh, nice! Thanks for the heads-up, @graysky2!

graysky2 commented:
Here is my putative attempt at it, again, need to check those "upstreamed" ones: https://github.com/graysky2/openwrt/commit/8c9405ef285e0fc27d562d99e914f96a9ea828b0

rsalvaterra commented:
@graysky2, I may be wrong, but I think generic/pending-5.10/498-mtd-mtdconcat-select-readwrite-function.patch should be dropped too. Analysing…

rsalvaterra committed:
kernel: bump 5.10 to 5.10.68

Deleted (upstreamed):
generic/backport-5.10/610-v5.13-02-netfilter-Fix-fall-through-warnings-for-Clang.patch
generic/backport-5.10/792-v5.15-0001-net-dsa-b53-Fix-calculating-number-of-switch-ports.patch
generic/backport-5.10/792-v5.15-0002-net-dsa-b53-Set-correct-number-of-ports-in-the-DSA-s.patch
generic/backport-5.10/792-v5.15-0003-net-dsa-b53-Fix-IMP-port-setup-on-BCM5301x.patch
generic/backport-5.10/840-0001-PCI-of-Don-t-fail-devm_pci_alloc_host_bridge-on-miss.patch
generic/backport-5.10/840-0002-PCI-iproc-Fix-BCMA-probe-resource-handling.patch
generic/pending-5.10/498-mtd-mtdconcat-select-readwrite-function.patch

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra commented:
> You might want to replace `platform` with the actual value. For example, `bcm27xx/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch`

Done too.

anomeome commented:
hmm, failure to launch on mvebu (mamba), finger pointing at the 498-mtd ???

rsalvaterra commented:
> hmm, failure to launch on mvebu (mamba), finger pointing at the 498-mtd ???

Ouch. Do you have a serial console to see what happened? :/

anomeome commented:
Right you be, [bootlog](https://termbin.com/awpi). I made a note where it stops dead, there will normally be a short pause at that point allowing one to enter a command. Proceeding with f continues but it is DIW. Checking things here...

rsalvaterra commented:
Hm… No issues on my Redmi AC2100… :/

anomeome commented:
Well, if it is me I am not spotting it, did a hard reset, make clean, applied just the minimal number of patches needed for my image, and no go. I have built / booted all iterations of this PR, currently running the .67 flavour on a  mamba and rango with no issue, I have not taken a kick at the rango with this .68 version.

anomeome commented:
Well with a _dirclean_ up and running on mvebu (mamba, rango), presumably [this](http://lists.openwrt.org/pipermail/openwrt-devel/2021-September/036352.html). Forgot about that note. might catch a few out I suppose.

rsalvaterra commented:
Oh, indeed! Not doing a complete clean after the musl bump is a recipe for disaster. Been there too. :)

rockdrilla commented:
Btw, 5.10.68 runs fine on ath79/generic (Archer C7 v2).

graysky2 commented:
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, ipq806x/R7800
Run-tested: bcm2711/RPi4B, ipq806x/R7800
```

adschm commented:
> Deleted (reverse-applicable):

Do you actually look at the patches and check whether they are upstreamed BTW? Being reverse-applicable is not a reason to remove patches, but just a reason to have a look. There might be parts that have not been upstreamed and other quicks.

So, the proper title would be "Deleted (upstreamed)" after this has been validated ...

regae commented:
failed to boot on rpi 3b+, 
reverted these commits, able to boot again
https://github.com/torvalds/linux/commit/1e7c57355a3bc617fc220234889e49fe722a6305
https://github.com/torvalds/linux/commit/09cbd1df7d2615c19e40facbe31fdcb5f1ebfa96

rsalvaterra commented:
Hi, @adschm,

> Do you actually look at the patches and check whether they are upstreamed BTW? Being reverse-applicable is not a reason to remove patches, but just a reason to have a look. There might be parts that have not been upstreamed and other quicks.

Actually, it's not the first time [you ask this](https://github.com/openwrt/openwrt/pull/4225#issuecomment-854895726), but the answer is "yes". ;)

> So, the proper title would be "Deleted (upstreamed)" after this has been validated ...

Yes, that makes sense. I'll reword and update the pull request. Thanks! :)

rsalvaterra commented:
> failed to boot on bcm2710/rpi3b+,
> reverted these commits, able to boot again
> [torvalds/linux@1e7c573](https://github.com/torvalds/linux/commit/1e7c57355a3bc617fc220234889e49fe722a6305)
> [torvalds/linux@09cbd1d](https://github.com/torvalds/linux/commit/09cbd1df7d2615c19e40facbe31fdcb5f1ebfa96)

Ouch, that's nasty. One wouldn't expect such problems to creep into supposedly stable (LTS!) kernels.

adschm commented on target/linux/generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch:
```diff
@@ -50,20 +50,14 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
 +	if (bus->dev) {
  		bcma_of_fill_device(bus->dev, core);
-+		dma_coerce_mask_and_coherent(dev, bus->dev->coherent_dma_mask);
++		dma_coerce_mask_and_coherent(&core->dev, bus->dev->coherent_dma_mask);
```
Are you sure the "&" is correct here? Previously is was assigned to `*dev`, but now it's replacing plain `dev`. I'd have expected just "core->dev" here. But I'm only looking at this with my limited C++ skills...

hauke replied:
Yes this is correct. 
The line above  does it in a similar way:
```
dev_set_name(&core->dev, "bcma%d:%d", bus->num, core->core_index);
```
`dev` is not a pointer in `core`, but directly embedded in the srtucture.

graysky2 commented:
5.10.69 is an auto-refresh only release :+1:

Boos4721 commented:
> 5.10.69 is an auto-refresh only release 👍

ya, no issues on Xiaomi AX3600
![](https://www.hualigs.cn/image/6150a7d8c9824.jpg)

rsalvaterra committed:
kernel: bump 5.10 to 5.10.69

Patches automatically refreshed.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

ldir-EDB0 commented:
Works for me(tm) on APU2 X86_64

graysky2 commented:
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, ipq806x/R7800
Run-tested: bcm2711/RPi4B, ipq806x/R7800
```

hauke commented:
> > failed to boot on bcm2710/rpi3b+,
> > reverted these commits, able to boot again
> > [torvalds/linux@1e7c573](https://github.com/torvalds/linux/commit/1e7c57355a3bc617fc220234889e49fe722a6305)
> > [torvalds/linux@09cbd1d](https://github.com/torvalds/linux/commit/09cbd1df7d2615c19e40facbe31fdcb5f1ebfa96)
> 
> Ouch, that's nasty. One wouldn't expect such problems to creep into supposedly stable (LTS!) kernels.
Is this problem with `bcm2710/rpi3b+` still there with kernel 5.10.69? 

How does the problem look like?
This is probably related to some of the extra patches we apply for bcm2710.

regae commented:
i will try .69 later, here is screenshot from .68

![Uploading 5503BE15-EB11-43CA-A8C3-B737D63FF034.jpeg…]()

rsalvaterra commented:
@regae, do you really need glibc? Musl is the obvious candidate for OpenWrt. However, that crashes happens way before anything even touches the libc.
@hauke, I don't see any recent changes to the RPi clock driver[[1]](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/drivers/clk/bcm/clk-bcm2835.c?h=linux-5.10.y)[[2]](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/drivers/clk/bcm/clk-raspberrypi.c?h=linux-5.10.y)… :/

rsalvaterra commented:
> failed to boot on bcm2710/rpi3b+, reverted these commits, able to boot again [torvalds/linux@1e7c573](https://github.com/torvalds/linux/commit/1e7c57355a3bc617fc220234889e49fe722a6305) [torvalds/linux@09cbd1d](https://github.com/torvalds/linux/commit/09cbd1df7d2615c19e40facbe31fdcb5f1ebfa96)

@regae, does reverting only https://github.com/gregkh/linux/commit/60831f5ae6c713afceb6d29f40899ed112f36059 fix the problem? (By the way, you linked to Linus' tree, OpenWrt uses the stable tree from gregkh.)

regae commented:
@rsalvaterra yes, removing those codes fixed my problem.
**still the same with .69**
reverting these two patches fixed it
_**firmware: raspberrypi: Fix a leak in 'rpi_firmware_get()'**_ 
_**firmware: raspberrypi: Keep count of all consumers**_

rsalvaterra commented:
> @rsalvaterra yes, removing those codes fixed my problem.
> **still the same with .69**
> reverting these two patches fixed it
> _**firmware: raspberrypi: Fix a leak in 'rpi_firmware_get()'**_
> _**firmware: raspberrypi: Keep count of all consumers**_

That's not what I asked. I asked if reverting **only** https://github.com/gregkh/linux/commit/60831f5ae6c713afceb6d29f40899ed112f36059 fixed the problem.

regae commented:
> > @rsalvaterra yes, removing those codes fixed my problem.
> > **still the same with .69**
> > reverting these two patches fixed it
> > _**firmware: raspberrypi: Fix a leak in 'rpi_firmware_get()'**_
> > _**firmware: raspberrypi: Keep count of all consumers**_
> 
> That's not what I asked. I asked if reverting **only** [gregkh/linux@60831f5](https://github.com/gregkh/linux/commit/60831f5ae6c713afceb6d29f40899ed112f36059) fixed the problem.

ok wait, will try, be back

rsalvaterra commented:
> edit: i think "**_firmware: raspberrypi: Fix a leak in 'rpi_firmware_get()'_**" is related to previous commit
> and of_find_device_by_node() is patched by 950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch

Indeed, there was an issue with that very same patch, I just updated the pull request. Can you give it a spin and see if it fixes the problem for you?

regae commented:
ok now its fixed, im able to boot .69
👍

rsalvaterra commented:
> ok now its fixed, im able to boot .69 👍
> 
> is it caused by put_device() ?

Yes. We use static `struct platform_device *g_pdev` to keep track of the platform device, while upstream keeps a refcount (`put_device`/`get_device`). By not removing those `put_device` calls, the refcount became unbalanced and that triggered the issue you saw.

rmandrad commented:
all good with mvebu/wrt3200acm

rsalvaterra committed:
kernel: bump 5.10 to 5.10.70

Deleted (upstreamed):
bcm53xx/patches-5.10/181-Revert-USB-bcma-Add-a-check-for-devm_gpiod_get.patch

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

ldir-EDB0 commented:
Still works on my APU2 X86_64

rmandrad commented:
can anyone tell me what is the release strategy for the kernel on openwrt ? we started at the  5.10.65 conversation and are now 5.10.70 ?

stintel reviewed (approved):
Build-and-runtime-tested on:
* ath79
* bcm27xx
  * bcm2708
  * bcm2709
  * bcm2711
* qoriq


Tested-by: Stijn Tintel <stijn@linux-ipv6.be>

ldir-EDB0 reviewed (approved):
looks good to me

Signed-off-by: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>

dangowrt commented:
Merged. Thanks!
